src/App.tsx

```diff
@@ -1,24 +1,78 @@
 import React from 'react';
 import './App.scss';
+import { Clock } from './component/Clock/Clock';
+import randomClockNames from './clockNames';
 
-const App: React.FC = () => {
-  const timerId: NodeJS.Timer = setInterval(() => {
-    const date: Date = new Date();
-
-    // eslint-disable-next-line
-    console.log(date.toLocaleTimeString());
-  }, 1000);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-      <p>
-        Current time:
-        {' '}
-        {/* Print the time here instead of DevTools */}
-      </p>
-    </div>
-  );
+type State = {
+  isVisible: boolean;
+  name: string;
 };
 
+class App extends React.Component<{}, State> {
+  state: State = {
+    isVisible: true,
+    name: 'Hello, gorgeous!',
+  };
+
+  changeName = () => {
+    const randomIndex = Math.floor(Math.random() * randomClockNames.length);
+
+    this.setState({ name: randomClockNames[randomIndex] });
+  };
+
+  showClock = () => {
+    this.setState({
+      isVisible: true,
+    });
+  };
+
+  hideClock = () => {
+    this.setState({
+      isVisible: false,
+    });
+  };
```

Comment:
you can create one method that set isClockVisible value

Reply:
That makes sense 🤔  Thanks